### PR TITLE
C++ Diamond Square code snippet added

### PR DESCRIPTION
**Pull request by SrijitaSarkar99**
## Diamond square in C++

Added markdown file for displaying diamond square pattern for value=n 
Related issue #22

### CODE REVIEW

File: snippets/c++/diamondSquare.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# Diamond Square
+
+## Code
+
+```C++
+int main()
+{
+    int n;
+    cout << "Enter the value of n : ";
```
Comment (kaustubhgupta) [inline snippets/c++/diamondSquare.md]: What is 'n' here? The number of rows, columns, or both? Whatever it is, comment out in the code

File: snippets/c++/diamondSquare.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            cout << "*";
+        cout << endl;
+    }
+    
+    return 0;
+}
+```
+
+## Output
+
```
Comment (kaustubhgupta) [inline snippets/c++/diamondSquare.md]: Attach the console output image instead of hardcoding this output

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

File: snippets/c++/diamondSquare.md (hunk 1 above)
Comment (SrijitaSarkar99) [reply snippets/c++/diamondSquare.md]: @kaustubhgupta I have made few changes. Please check.
n is the number of rows as mentioned in the problem statement

File: snippets/c++/Diamond_Square/diamondSquare.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        cout << endl;
+    }
+    return 0;
+}
+```
+
+## Output
+
+![Diamond image](row_1.png?raw=true "Diamond(row=1)")  
+![Diamond image](rows_6.png?raw=true "Diamond(row=6)")
```
Comment (sakshamarora1) [inline snippets/c++/Diamond_Square/diamondSquare.md]: Only a single output file will suffice. Let's keep the 6 rows output.

**Review by sakshamarora1 — CHANGES_REQUESTED** [review]

File: snippets/c++/diamondSquare.md (hunk 1 above)
Comment (kaustubhgupta) [reply snippets/c++/diamondSquare.md]: I know n is the number of rows but how will someone who is viewing the code will get it? That's why it was suggested and now you have made the change, it's fine now

File: snippets/c++/Diamond_Square/diamondSquare.md (hunk 3 above)
Comment (SrijitaSarkar99) [reply snippets/c++/Diamond_Square/diamondSquare.md]: @sakshamarora1 I have made the changes. Please check

**Review by sakshamarora1 — APPROVED** [review]

**Review by kaustubhgupta — APPROVED** [review]

**Pull request merged**